CMakeLists.txt

```diff
@@ -974,4 +981,65 @@ else()
             endif()
         endif()
     endif()
+
+    if ((BUILD_LINUX_SHARED_LIB STREQUAL "ON") AND (CMAKE_SYSTEM_NAME MATCHES "Linux"))
+        SET(LIB_NAME "spdm")
+        SET(CMAKE_PROJECT_NAME "libspdm")
+        SET(CMAKE_PROJECT_DESCRIPTION "libspdm shared library")
+
+        SET(CMAKE_SKIP_BUILD_RPATH FALSE)
+        SET(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
+        SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
+        SET(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+
+        add_library(${LIB_NAME} SHARED 
+            $<TARGET_OBJECTS:spdm_common_lib>
+            $<TARGET_OBJECTS:spdm_crypt_lib>
+            $<TARGET_OBJECTS:spdm_requester_lib>
+            $<TARGET_OBJECTS:spdm_responder_lib>
+            $<TARGET_OBJECTS:spdm_secured_message_lib>
+            $<TARGET_OBJECTS:spdm_transport_mctp_lib>
+            $<TARGET_OBJECTS:spdm_transport_pcidoe_lib>
+        )
+
+        add_library(${LIB_NAME}_platform SHARED 
+            $<TARGET_OBJECTS:debuglib>
+            $<TARGET_OBJECTS:malloclib>
+            $<TARGET_OBJECTS:platform_lib>
```

Comment:
FYI : the [watchdog timers](https://github.com/DMTF/libspdm/blob/main/os_stub/platform_lib/watchdog.c) currently don't do anything, so you should file an issue to have these implemented on Linux.

Reply:
I created an issue for this, thank you.